xds/internal/xdsclient/xdsresource/resource_type.go

```diff
@@ -25,10 +25,20 @@
 package xdsresource
 
 import (
+	"google.golang.org/grpc/xds/internal"
 	"google.golang.org/grpc/xds/internal/xdsclient/bootstrap"
+	"google.golang.org/grpc/xds/internal/xdsclient/xdsresource/version"
 	"google.golang.org/protobuf/types/known/anypb"
 )
 
+func init() {
+	internal.ResourceTypeMapForTesting = make(map[string]interface{})
+	internal.ResourceTypeMapForTesting[version.V3ListenerURL] = listenerType
+	internal.ResourceTypeMapForTesting[version.V3RouteConfigURL] = routeConfigType
+	internal.ResourceTypeMapForTesting[version.V3ClusterURL] = clusterType
+	internal.ResourceTypeMapForTesting[version.V3EndpointsURL] = endpointsType
+}
+
 // Producer contains a single method to discover resource configuration from a
```

Comment:
This diff is not part of this PR. This would be added after this [PR](https://github.com/grpc/grpc-go/pull/5996) is merged in.